addons/account_facturx/models/account_move.py

```diff
@@ -97,14 +97,14 @@ def _import_facturx_invoice(self, tree):
         '''
         def find_partner(partner_type):
             elements = tree.xpath('//ram:%s/ram:SpecifiedTaxRegistration/ram:ID' % partner_type, namespaces=tree.nsmap)
-            partner = elements and self.env['res.partner'].search([('vat', '=', elements[0].text)], limit=1)
+            partner = elements and self.env['res.partner'].search([('vat', '=', elements[0].text)], limit=1) or self.env["res.partner"]
             if not partner:
                 elements = tree.xpath('//ram:%s/ram:Name' % partner_type, namespaces=tree.nsmap)
                 partner_name = elements and elements[0].text
-                partner = elements and self.env['res.partner'].search([('name', 'ilike', partner_name)], limit=1)
+                partner = elements and self.env['res.partner'].search([('name', 'ilike', partner_name)], limit=1) or partner
             if not partner:
                 elements = tree.xpath('//ram:%s//ram:URIID[@schemeID=\'SMTP\']' % partner_type, namespaces=tree.nsmap)
-                partner = elements and self.env['res.partner'].search([('email', '=', elements[0].text)], limit=1)
+                partner = elements and self.env['res.partner'].search([('email', '=', elements[0].text)], limit=1) or partner
             return partner
```

Comment:
Why not simply modifying this last line and do return partner or self.env['res.partner'] ? Would be cleaner than adding all these 'or' imo.

Reply:
Indeed! Sometimes you start unraveling the skein, and don't see that the final result can be simplified. Changed.